### PR TITLE
feat(asset-server-plugin): optimize S3 file handling

### PR DESCRIPTION
**Pull request by asonnleitner**
- Added Buffer import from 'node:buffer'.
- Replaced manual buffer concatenation with `transformToByteArray` for efficiency in `toBuffer`.
- Improved `readFileToStream` by utilizing `Readable.from` for the byte array.

These changes enhance the performance and readability of S3 file handling operations.

### CODE REVIEW

File: packages/asset-server-plugin/src/s3-asset-storage-strategy.ts [hunk 1]
```diff
@@ -230,14 +226,15 @@ export class S3AssetStorageStrategy implements AssetStorageStrategy {
             });
         }
 
-        return body;
+        const typedArray = await body.transformToByteArray();
+        return Readable.from(typedArray)
```
Comment (michaelbromley) [inline packages/asset-server-plugin/src/s3-asset-storage-strategy.ts:230]: In this part, as far as I understand you are "consuming" the stream into a buffer and then creating a new stream from that buffer. That seems redundant to me? Or am I missing something?

**Review by dfernandesbsolus — APPROVED** [review]

File: packages/asset-server-plugin/src/s3-asset-storage-strategy.ts [hunk 2]
```diff
@@ -210,13 +211,8 @@ export class S3AssetStorageStrategy implements AssetStorageStrategy {
...
             return Buffer.from('');
         }
-
-        const chunks: Buffer[] = [];
-        for await (const chunk of body) {
-            chunks.push(chunk);
-        }
-
-        return Buffer.concat(chunks);
+        const typedArray = await body.transformToByteArray()
```
Comment (dfernandesbsolus) [inline packages/asset-server-plugin/src/s3-asset-storage-strategy.ts:214]: Nice!

**Review by dfernandesbsolus — APPROVED** [review]

**Review by michaelbromley — COMMENTED** [review]
Thanks for the contribution! Check my comment and let me know what you think.

**Comment by michaelbromley** [comment]
Hi, I'm going to close this for now because the implementation was unclear to me, as per my comment above. Open to a new PR which addresses this. Thanks for your time 👍

**Pull request closed without merging**